src/whylogs/app/logger.py

```diff
@@ -591,8 +591,12 @@ def log_segments_keys(self, data):
 
         for each_segment in segments:
             try:
-                segment_df = grouped_data.get_group(each_segment)
-                segment_tags = [{"key": self.segments[i], "value": each_segment[i]} for i in range(len(self.segments))]
+                if len(self.segments) == 1:
```

Comment:
Looks good Felipe! We should consider adding a test for the single dictionary entry case, since we had a bug in this pattern, but I understand your changes work for both single key and single dictionary entry, so let's do that in a follow up change.

e.g. a test that passes something like this:
```
test_segments = [
        [{"key": "home_ownership", "value": "RENT"}],
    ]
```